send.go

```diff
@@ -71,32 +72,40 @@ func Send(target net.UDPAddr, data []byte, timeout time.Duration) (time.Duration
 	return sendThroughConnection(connection, target, data)
 }
 
-func SendIndefinitely(target net.UDPAddr, data []byte, timeout time.Duration, interval time.Duration, stop chan bool) []time.Duration {
+type PackageSlip struct {
+	Target   net.UDPAddr
+	Data     []byte
+	Timeout  time.Duration
+	Interval time.Duration
+}
+
+func SendIndefinitely(ctx context.Context) []time.Duration {
+	slip := ctx.Value("packageSlip").(PackageSlip)
```

Comment:
As i mention above, this introduces new issue.  If key `"packageSlip"` is not found, this will panic.  Also if the value in `packageSlip` is not of the asserted type `PackageSlip` you also may get a panic there.  The way to fix that is 
  a) dont pass package slip inside context, as mentioned above, or
  b) use the long form of type assertion to ensure the assertion dont panic

```go
slip, ok := ctx.Value("packageSlip).(PackageSlip)

// ok is false if assertion cant be done, handle failure here
if !ok { // }
```